File: app/src/main/java/igrek/todotree/services/history/change/AddItemItemChange.java
package igrek.todotree.services.history.change;


import igrek.todotree.model.treeitem.AbstractTreeItem;

public class AddItemItemChange extends AbstractItemChange {
	
	private AbstractTreeItem newItem;
	private AbstractTreeItem parent;
	private int insertPosition;
	
	public AddItemItemChange(AbstractTreeItem newItem, AbstractTreeItem parent, int insertPosition) {
		this.newItem = newItem;
		this.parent = parent;
		this.insertPosition = insertPosition;
	}
	
	public AbstractTreeItem getNewItem() {
		return newItem;
	}
	
	public AbstractTreeItem getParent() {
		return parent;
	}
	
	public int getInsertPosition() {
		return insertPosition;
	}
	
	@Override
	public void revert() {
		//TODO
	}
}

File: app/src/main/java/igrek/todotree/service/history/change/AddItemItemChange.kt
package igrek.todotree.service.history.change

import igrek.todotree.domain.treeitem.AbstractTreeItem
import igrek.todotree.service.history.change.AbstractItemChange

class AddItemItemChange(
    val newItem: AbstractTreeItem,
    val parent: AbstractTreeItem,
    val insertPosition: Int
) : AbstractItemChange() {
    override fun revert() {
        //TODO
    }
}